features/integration/applications.feature

```diff
@@ -125,15 +126,79 @@ Feature: Applications
+      And I sign and submit the transaction, saving the txid. If there is an error it is "".
+      And I wait for the transaction to be confirmed.
+      Then according to "algod", the contents of the box with name "b64:MTE0NTE0" in the current application should be "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA". If there is an error it is "".
+
+      # set box "str:foo bar" to value "str:baz qux"
+      And I build an application transaction with the transient account, the current application, suggested params, operation "call", approval-program "", clear-program "", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "str:set,str:foo bar,str:baz qux", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0, boxes "0,str:foo bar"
+      And I sign and submit the transaction, saving the txid. If there is an error it is "".
+      And I wait for the transaction to be confirmed.
+      Then according to "algod", the contents of the box with name "str:foo bar" in the current application should be "YmF6IHF1eAAAAAAAAAAAAAAAAAAAAAAA". If there is an error it is "".
+
+      # full check confirmed by both algod and indexer
+      Then according to "algod", the current application should have the following boxes "str:foo bar,b64:APj/IA==,str:name,b64:MTE0NTE0".
+      And according to "algod", by parameter max 2, the current application should have 2 boxes.
+      And according to "algod", by parameter max 6, the current application should have 4 boxes.
+      And I sleep for 1 seconds for indexer to digest things down.
+      And according to "indexer", the current application should have the following boxes "str:foo bar,b64:APj/IA==,str:name,b64:MTE0NTE0".
+      And according to "indexer", by parameter max 2, the current application should have 2 boxes.
+      And according to "indexer", by parameter max 6, the current application should have 4 boxes.
+
+      # delete one box
+      Then I build an application transaction with the transient account, the current application, suggested params, operation "call", approval-program "", clear-program "", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "str:delete,str:name", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0, boxes "0,str:name"
+      And I sign and submit the transaction, saving the txid. If there is an error it is "".
+      And I wait for the transaction to be confirmed.
+      And according to "algod", the current application should have the following boxes "str:foo bar,b64:APj/IA==,b64:MTE0NTE0".
+
+      # move to indexer testing steps
+      And I sleep for 1 seconds for indexer to digest things down.
```

Comment:
🤣

Reply:
well, one finding I realized during the testcase writing is that, *a certain wait interval* is needed, for no waiting leads to indexer fail to reflect changes applied on algod side (e.g., deletion).

But indeed, the interval may be even shorter than 1 sec. My thinking is: this might be good nuf? For I am concerned that testing will be flaky if we be aggressive and trim down even more... I confess that this is a quick dirty trick.

Reply:
Given your comment, I recommend that you change the step to `I sleep for {int} milliseconds for indexer to digest things down.`

This makes it a bit more future proof. We can start with a small number like 500 and then increase if we find tests are failing.